shared/settings.cpp

```diff
@@ -0,0 +1,102 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <QDebug>
+#include "settings.h"
+
+char* readFile(const char* path) {
+    FILE *file = fopen(path, "r");
+    if (file) {
+        fseek(file, 0, SEEK_END);
+        long length=ftell(file);
+        fseek (file, 0, SEEK_SET);
+        char *buffer = (char*)malloc(length+1);
+        if (buffer) {
+            fread (buffer, 1, length, file);
+            buffer[length] = '\0';
+        }
+        fclose(file);
+        return buffer;
+    }
+    return nullptr;
+}
+
+bool doesStartWithPrefixes(const char* const prefixes[], int arrLength, char* toCheck) {
+    int length = strlen(toCheck);
+    for (int i = 0; i < arrLength; ++i) {
+        int prefixLength = strlen(prefixes[i]);
+        if (prefixLength < length && strncmp(prefixes[i], toCheck, prefixLength) == 0) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Settings::readDeviceType() {
+    char* device_type = readFile("/sys/devices/soc0/machine");
+    _deviceType = DeviceType::UNKNOWN;
+
+    if (device_type) {
+
+        const char* const remarkable1[] {
```

Comment:
We don't know if these are the only two valid values. I'd prefer to assume we are on the rM1 if we don't detect the rM2. If the file doesn't exist we can then say it's unknown.